pkg/skaffold/jib/jib.go

```diff
@@ -17,64 +17,173 @@ limitations under the License.
 package jib
 
 import (
+	"encoding/json"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"sort"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/karrick/godirwalk"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func getDependencies(cmd *exec.Cmd) ([]string, error) {
-	stdout, err := util.RunCmdOut(cmd)
-	if err != nil {
-		return nil, err
+// pathMap contains fields parsed from Jib's JSON output
+type pathMap struct {
+	// Build lists paths to build definitions that trigger a call out to Jib to refresh the pathMap, as well as a rebuild, upon changing
+	Build []string `json:"build"`
+
+	// Inputs lists paths to build dependencies that trigger a rebuild upon changing
+	Inputs []string `json:"inputs"`
+
+	// Results lists paths to files that should be ignored when checking for changes to rebuild
+	Results []string `json:"ignore"`
```

Comment:
Why not call it `Ignored`?

Reply:
See this thread: https://github.com/GoogleContainerTools/skaffold/pull/1744#discussion_r274109892

Reply:
OK. I found it super hard to understand

Reply:
I also find this a bit hard to understand. @briandealwis's comment makes sense that they're not being ignored by the file syncer, but in this context isn't this list part of the *build* dependencies, and are being ignored by the *builder*?

I don't feel super strongly about it since it's just a name, but something to consider. maybe the comment could clarify this just a bit.

Reply:
This structure describes different categories or types of files and directories found in an artifact's workingdir (`context`).  the watcher and filesync react to changes to specific categories of files/directories. The watcher, for example,  reacts only to changes in the _build definitions_ and _source inputs_, an ignores changes to _build results_ — there's no point triggering a rebuild due to a change in a build result.  Whereas filesync should react to the union of _build definition_, _source inputs_, and _build results_.

Maybe it makes more sense to make the categories more purpose-driven. Instead of  having filesync use the union of several categories, we'd merge them into a single category _syncable_. And also have separate categories for _reload_ and _rebuild_?